### PR TITLE
Use order language for Invoice and deliveryslip

### LINKED ISSUE #28712: Customer language not in invoice or delivery slip in BO
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Wrong language for customer's delivery slips/invoices

There is a super easy bug fix posted [on the forums](https://www.prestashop.com/forums/topic/487202-invoice-language/) 

We literally only have to add one single line to /classes/pdf/HTMLTemplateInvoice.php:
In the function public function __construct, search for :
 ```
// header informations
$this->date = Tools::displayDate($order_invoice->date_add);
And insert below:
`Context::getContext()->language = new Language($this->order->id_lang);`
```

Maybe somebody can quickly do this. I'm not as experienced with GitHub yet.

### Expected behavior

Show invoice/delivery slips pdf in customer's language

### Steps to reproduce

1. Go to BO > International > Localization
2. Add multiple languages to your shop
3. Go to FO > make some orders with different languages
4. Go to BO > Orders > Orders page > Click on orders concerned to view details
5. Change status to "delivered" for example
6. Click on invoice and/or delivery slip button to download pdf files
7. See error, language has the admin's language.

### PrestaShop version(s) where the bug happened

1.7.8.6

### PHP version(s) where the bug happened

8.4

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add options to use or not  order language for Invoice and deliveryslip
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/28712
| Related PRs       | 
| How to test?      | Make an order in french, go to the BO set your language in english and check if the invoice is in french 
| Possible impacts? |

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

File: src/PrestaShopBundle/Form/Admin/Sell/Order/Delivery/SlipOptionsType.php [hunk 1]
```diff
@@ -74,6 +74,17 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         'Admin.Orderscustomers.Help'
                     ),
                 ]
+            )
+            ->add(
+                'deliveryslip_language',
+                SwitchType::class,
+                [
+                    'label' => $this->trans('Delivery slip Language', 'Admin.Orderscustomers.Feature'),
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Order/Delivery/SlipOptionsType.php]:
```suggestion
                    'label' => $this->trans('Language of the delivery slip', 'Admin.Orderscustomers.Feature'),
```

File: src/PrestaShopBundle/Form/Admin/Sell/Order/Delivery/SlipOptionsType.php [hunk 2]
```diff
@@ -74,6 +74,17 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                    'help' => $this->trans(
+                        'Use the language of the order for the delivery slip.',
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Order/Delivery/SlipOptionsType.php]:
```suggestion
                        'Create a delivery slip in the same language as the one used for the order.',
```

File: src/PrestaShopBundle/Form/Admin/Sell/Order/Invoices/InvoiceOptionsType.php [hunk 3]
```diff
@@ -107,7 +107,18 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'choices' => $this->invoiceModelChoiceProvider->getChoices(),
                 'translation_domain' => false,
             ])
-            ->add('use_disk_cache', SwitchType::class);
+            ->add('use_disk_cache', SwitchType::class)
+            ->add(
+                'invoice_language',
+                SwitchType::class,
+                [
+                    'label' => $this->trans('Invoice language', 'Admin.Orderscustomers.Feature'),
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Order/Invoices/InvoiceOptionsType.php]:
```suggestion
                    'label' => $this->trans('Language of the invoice', 'Admin.Orderscustomers.Feature'),
```

File: src/PrestaShopBundle/Form/Admin/Sell/Order/Invoices/InvoiceOptionsType.php [hunk 4]
```diff
@@ -107,7 +107,18 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                    'help' => $this->trans(
+                        'Use the language of the order for the invoice.',
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Order/Invoices/InvoiceOptionsType.php]:
```suggestion
                        'Create an invoice in the same language as the one used for the order.',
```

File: src/Adapter/Order/Delivery/SlipOptionsConfiguration.php [hunk 5]
```diff
@@ -59,6 +60,7 @@ public function getConfiguration()
             'prefix' => $this->configuration->get(self::PREFIX),
             'number' => $this->configuration->getInt(self::NUMBER),
             'enable_product_image' => $this->configuration->getBoolean(self::ENABLE_PRODUCT_IMAGE),
+            'deliveryslip_language' => $this->configuration->getBoolean(self::DELIVERY_LANGUAGE),
```
Comment (Amoifr) [inline src/Adapter/Order/Delivery/SlipOptionsConfiguration.php]: Just a detail ... is that name can be edited : "deliveryslip_language" ? or it's legacy and can cause a BC break ?
To respect the snake case convention it should be "delivery_slip_language".

Comment (ghost) [reply src/Adapter/Order/Delivery/SlipOptionsConfiguration.php]: Nice, i updated the PR

**Review by MeKeyCool — APPROVED** [review]

**Comment by Hlavtox** [comment]
@okom3pom Guys do you think we can somehow refactor the PDF renderer to use a custom language we tell it to?

This context language changing is smelly :-D

I tried to start something here - https://github.com/PrestaShop/PrestaShop/pull/26234/files

**Comment by ghost** [comment]
@Hlavtox 

I agree with you! But we can't make BC before 9.0.0 this improvement will be useful for 8.x

For example [this PR](https://github.com/PrestaShop/PrestaShop/pull/28931) will be for 9.0.0 and I will make another simple without local language  for 8.1.0 

I am not sure where the PRs will go in the develop branch?

I ok to try an other PR but for wich version?

**Review by kpodemski — CHANGES_REQUESTED** [review]
I wanted to go for ✅, but I noticed a few conflicts. I think this PR can be planned for v8.1, so maybe you could rebase it?

I know it's frustrating, but a list of PRs to check is quite long :(

**Comment by hanuchin** [comment]
Are there any other solutions / commits for this issue/PR?

We want to use PrestaShop 8.1.6 but we are not able to use the order.id_lang for the pdf creation. 
The pdf language is depending on the backoffice employee language. 

This is a major issue for us.

**Pull request closed without merging**